library/CHART-TEMPLATE/templates/sidecar-agent-configmap.yaml

```diff
@@ -0,0 +1,6 @@
+{{- include "common.agentConfig.configMap" (dict
+    "global"  .Values.global.agent.config
+    "local"   .Values.agent.config
+    "name"    (printf "%s-example-config" (include "common.naming.fullname" . ))
```

Comment:
Ok, I see now: It's because we might want to have different agent `configMap`s. My question would be: Does it make sense then? All the other helpers are built around centralizing common things, while this one needs to do advanced logic to be able to be too flexible. Couldn't that fact be telling us that the agent config is not as `common` as we thought?

Reply:
Ok, I see now: It's because we might want to have different agent `configMap`s. My question would be: Does it make sense then? All the other helpers are built around centralizing common things, while this one needs to do advanced logic to be able to be too flexible. Couldn't that fact be telling us that the agent config is not as `common` as we thought?

Reply:
Images [has also this complex behavior](https://github.com/newrelic/helm-charts/blob/master/library/common-library/templates/_images.tpl#L17-L30) that [leads to complex usage](https://github.com/newrelic/helm-charts/blob/master/library/CHART-TEMPLATE/templates/deployment.yaml#L22-L25).

When there is "more than one" either image or agent it leads to this pattern because Helm is not flexible enough.

Although it could look complex, it uses the same pattern as Bitnami.